### ISSUE TITLE
Node/Mongo docker-compose's mongoDB container should be available to the local host

### ISSUE THREAD

**Issue by diberry** [microsoft/vscode-dev-containers#506]
The mongo container should be available externally or at least give me the commented out docker-compose settings. This dev container is supposed to help me debug but isn't helpful if I can't see what eventually gets in the DB.

The container should also work for windows/linux/mac. Windows is picky about the host for a mongo container.

**Comment by anthonychu** [comment]
Another option to think about is to add the MongoDB extension to the devcontainer. That way the user can browse and manage the DB even without tools installed locally on their machine.

If Codespaces eventually supports compose, this would also be helpful there as the extension is available in the in-browser editor.

**Comment by Chuxel** [comment]
>The container should also work for windows/linux/mac. Windows is picky about the host for a mongo container.

Can you be more specific? To be clear, MongoDB is in a separate container, so localhost inside the web container won't work on any platform if that is what you are referring to.  It's like having two machines sitting next to each other.

> Another option to think about is to add the MongoDB extension to the devcontainer. That way the user can browse and manage the DB even without tools installed locally on their machine.

I'd tend to prefer this option rather than exposing a port by default without the developer knowing about it.

> If Codespaces eventually supports compose, this would also be helpful there as the extension is available in the in-browser editor.

It actually does today!

**Comment by anthonychu** [comment]
> It actually does today!

I have clearly missed the memo. This is great news!

**Comment by diberry** [comment]
On the dev container for mongodb/node, neither the mongodb or node app are available to the local computer because no ports are open. The only availability is from within an interactive terminal (or vscode debugging of that terminal). 

Who is the main customer of this docker-compose setup? 
Why not show the port syntax (commented out) in the docker-compose file for both web and db?

**Comment by diberry** [comment]
Port forwarding works - I didn't see it in the .devcontainer.json file - can it be added but commented out? Along with the note that it can be done temporarily from the command bar?

**Comment by diberry** [comment]
How do I forward ports in the .devcontainer.json file but for aach service in the docker-compose?

**Comment by Chuxel** [comment]
Yep, I added it into the devcontainer.json when you raised your earlier issue since it actually was supposed to be there. It's in pretty much every other definition - so it, just got missed here for some reason. ☹️  Sorry about that!

Can also review extensions to see if there's one to include by default. Database ones have lower usage numbers but the Azure DB extension does support straight mongo as well.  Need to think through that a bit.
